tcms/testplans/static/testplans/js/get.js

```diff
@@ -34,8 +36,91 @@ $(document).ready(function() {
     });
 
     collapseDocumentText();
+    initTestCaseSearchAndAdd(testPlanId);
 });
 
+
+function addTestCaseToPlan(planId) {
+    const case_name = $('#search-testcase')[0].value;
+    const test_case = autocomplete_cache[case_name];
+
+    // test case is already present so don't add it
+    if ( allTestCases[test_case.id] !== undefined ) {
+        $('#search-testcase').val('');
+        return false;
+    }
+
+    jsonRPC('TestPlan.add_case', [planId, test_case.id], function(result) {
+        // IMPORTANT: the API result includes a 'sortkey' field value!
+        window.location.reload(true);
+
+        // TODO: remove the page reload above and add the new case to the list
+        // NB: pay attention to drawTestCases() & treeViewBind()
+        // NB: also add to allTestCases !!!
+
+        $('#search-testcase').val('');
+    });
+}
+
+
+function initTestCaseSearchAndAdd(planId) {
+    // + button
+    $('#btn-add-case').click(function() {
+        addTestCaseToPlan(planId);
+
+        return false;
+    });
+
+    // Enter key
+    $('#search-testcase').keyup(function(event) {
+        if (event.keyCode === 13) {
+            addTestCaseToPlan(planId);
+
+            return false;
+        };
+    });
+
+    // autocomplete
+    $('#search-testcase.typeahead').typeahead({
+        minLength: 1,
+        highlight: true
+        }, {
+        name: 'testcases-autocomplete',
+        // will display up to X results even if more were returned
+        limit: 100,
+        async: true,
+        display: function(element) {
+            const display_name = 'TC-' + element.id + ': ' + element.summary;
+            autocomplete_cache[display_name] = element;
+            return display_name;
```

Comment:
string concatenation is just ugly
```suggestion
            const displayName = `TC-${element.id}: ${element.summary};
            autocomplete_cache[displayName] = element;
            return displayName;
```

Reply:
Suggestion missing closing tickmark